### PR TITLE
fix: type ResultMetaData fields as optional

### LINKED ISSUE #1423: ResultMetadata fields not typed as Optional
Please answer these questions before submitting your issue. Thanks!

1. What version of Python are you using?

  python 3.9

2. What operating system and processor architecture are you using?

   macos

3. What are the component versions in the environment (`pip freeze`)?

   snowflake-connector-python==3.0.0

4. What did you do?

   ResultMetadata.precision, ResultMetadata.display_size, ResultMetadata,internal_size, ResultMetadata.precision, ResultMetadata.scale can be `None` but aren't typed as such, so the type checker (pyright) cannot provide static checks that the `None` is handled. 

5. What did you expect to see?

   Where a ResultMetadata field can be `None` this is reflected in the type signature.

### PR DESCRIPTION
**Pull request by tekumara**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #1423

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am modifying authorization mechanisms
   - [ ] I am adding new credentials
   - [ ] I am modifying OCSP code
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   type ResultMetaData fields as optional where they can be `None`

### CODE REVIEW

**Comment by tekumara** [comment]
@sfc-gh-mkeller would you be able to take a look at this? 🙏

**Review by sfc-gh-jdu — APPROVED** [review]

File: src/snowflake/connector/cursor.py [hunk 1]
```diff
@@ -120,10 +121,10 @@
...
     name: str
     type_code: int
-    display_size: int
-    internal_size: int
-    precision: int
-    scale: int
+    display_size: Optional[int]
+    internal_size: Optional[int]
+    precision: Optional[int]
+    scale: Optional[int]
```
Comment (sfc-gh-mkeller) [inline src/snowflake/connector/cursor.py:127]: We should be using future annotations
```suggestion
    display_size: int | None
    internal_size: int | None
    precision: int | None
    scale: int | None
```

**Review by sfc-gh-mkeller — CHANGES_REQUESTED** [review]

**Comment by tekumara** [comment]
@sfc-gh-mkeller please see the changes I've made above

**Review by sfc-gh-mkeller — APPROVED** [review]
:ship: :ship:

**Pull request merged**